### PR TITLE
SitConfig and ItMonitoring mysql config changes

### PR DESCRIPTION
**Pull request by marinakog**

### CODE REVIEW

**Comment by marinakog** [comment]
http://wls-jenkins.us.oracle.com/job/weblogic-kubernetes-operator-javatest/3225 jerkin passed for these tests.
http://wls-jenkins.us.oracle.com/job/weblogic-kubernetes-operator-javatest/3224/

**Comment by sankarpn** [comment]
Marina, The changes look good, please resolve the conflicts.

**Comment by marinakog** [comment]
my tests are clean in full and partial  runs  http://wls-jenkins.us.oracle.com/job/weblogic-kubernetes-operator-javatest/3262   , http://wls-jenkins.us.oracle.com/job/weblogic-kubernetes-operator-javatest/3254/

File: integration-tests/src/test/java/oracle/kubernetes/operator/ItSitConfigDomainInImage.java [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+  private static String configOverrideDir = "";
+  private static String mysqlYamlFile = "";
+  private static String domainNS;
+  private static String testprefix = "sitconfigdomaininimage";
+  private static String mysqldbport;
+  private static String adminpodname;
+  private static String JDBC_URL;
+  private static StringBuffer namespaceList;
+  /**
+   * This method gets called only once before any of the test methods are executed. It does the
```
Comment (vanajamukkara) [inline integration-tests/src/test/java/oracle/kubernetes/operator/ItSitConfigDomainInImage.java]: modify the description based on the new changes 

  *  This method gets called only once before any of the test methods are executed. It does the
   * initialization of the integration test properties defined in OperatorIT.properties.
   *
   * @throws Exception when the initialization fails.

File: integration-tests/src/test/java/oracle/kubernetes/operator/ItSitConfigDomainInImage.java [hunk 2]
```diff
@@ -0,0 +1,277 @@
...
+      testClassName = new Object() {
+      }.getClass().getEnclosingClass().getSimpleName();
+      initialize(APP_PROPS_FILE, testClassName);
+      TEST_RES_DIR = getProjectRoot() + "/integration-tests/src/test/resources/";
+      testNumber = getNewSuffixCount();
+    }
+  }
+
+  /**
+   * This method gets called only once before any of the test methods are executed. It does the
```
Comment (vanajamukkara) [inline integration-tests/src/test/java/oracle/kubernetes/operator/ItSitConfigDomainInImage.java]: * This method gets called before every test. It creates the resultRoot, pvRoot directories, creates operator and 
* domain if not running.
   *
   * @throws Exception if results/pv directory or operator or domain creation fails.
   */

File: integration-tests/src/test/java/oracle/kubernetes/operator/ItSitConfigDomainInImage.java [hunk 3]
```diff
@@ -0,0 +1,277 @@
...
+   * initialization of the integration test properties defined in OperatorIT.properties and setting
+   * the resultRoot, pvRoot and projectRoot attributes.
+   *
+   * @throws Exception when the initialization, creating directories , copying files and domain
+   *                   creation fails.
+   */
+  @Before
+  public void prepare() throws Exception {
```
Comment (vanajamukkara) [inline integration-tests/src/test/java/oracle/kubernetes/operator/ItSitConfigDomainInImage.java]: The code seems to be same for both the classes, can u move this to super class by adding a new method with args and call that method here

File: integration-tests/src/test/java/oracle/kubernetes/operator/ItSitConfigDomainInPV.java [hunk 4]
```diff
@@ -0,0 +1,323 @@
...
+  private static String mysqlYamlFile = "";
+  private static String domainNS;
+  private static String testprefix = "sitconfigdomaininpv";
+  private static String mysqldbport;
+  private static String ADMINPODNAME;
+  private static String JDBC_URL;
+  private static StringBuffer namespaceList;
+
+  /**
+   * This method gets called only once before any of the test methods are executed. It does the
```
Comment (vanajamukkara) [inline integration-tests/src/test/java/oracle/kubernetes/operator/ItSitConfigDomainInPV.java]: same comments as for the other ItSitConfig class.

File: integration-tests/src/test/java/oracle/kubernetes/operator/ItSitConfigDomainInImage.java [hunk 5]
```diff
@@ -0,0 +1,277 @@
...
+        namespaceList.append(" ").append(domainNS);
+        sitconfigTmpDir = getResultDir() + "/sitconfigtemp" + testprefix;
+        mysqltmpDir = sitconfigTmpDir + "/mysql";
+        configOverrideDir = sitconfigTmpDir + "/configoverridefiles";
+        mysqlYamlFile = mysqltmpDir + "/mysql-dbservices.yml";
+
+        Files.createDirectories(Paths.get(sitconfigTmpDir));
+        Files.createDirectories(Paths.get(configOverrideDir));
+        Files.createDirectories(Paths.get(mysqltmpDir));
+        mysqldbport = String.valueOf(31306 + testNumber);
```
Comment (vanajamukkara) [inline integration-tests/src/test/java/oracle/kubernetes/operator/ItSitConfigDomainInImage.java]: create a constant variable for mysql db base port 31306 ...best place to generate port numbers is BaseTest.createDomainMap() but they may be too many changes for you ...

File: integration-tests/src/test/java/oracle/kubernetes/operator/ItSitConfigDomainInImage.java (hunk 1 above)
Comment (marinakog) [reply integration-tests/src/test/java/oracle/kubernetes/operator/ItSitConfigDomainInImage.java]: fixed

File: integration-tests/src/test/java/oracle/kubernetes/operator/ItSitConfigDomainInImage.java (hunk 2 above)
Comment (marinakog) [reply integration-tests/src/test/java/oracle/kubernetes/operator/ItSitConfigDomainInImage.java]: fixed

File: integration-tests/src/test/java/oracle/kubernetes/operator/ItSitConfigDomainInPV.java (hunk 4 above)
Comment (marinakog) [reply integration-tests/src/test/java/oracle/kubernetes/operator/ItSitConfigDomainInPV.java]: fixed

File: integration-tests/src/test/java/oracle/kubernetes/operator/ItSitConfigDomainInImage.java (hunk 5 above)
Comment (marinakog) [reply integration-tests/src/test/java/oracle/kubernetes/operator/ItSitConfigDomainInImage.java]: mysql is independent from domain creation, will keep separate, port number is generated as 31306 + getNewSuffixNumber()

**Comment by marinakog** [comment]
full test passed http://wls-jenkins.us.oracle.com/job/weblogic-kubernetes-operator-javatest/3290

**Pull request merged**